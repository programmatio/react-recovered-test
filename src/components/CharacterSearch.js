import React from "react";
import PropTypes from "prop-types";

const CharacterSearch = ({ name, handleNameUpdate, handleFetchCharacters }) => {
  return (
    <div>
      <form onSubmit={e => handleFetchCharacters(e)}>
        <label>
          Search Starwars Character by Name:
          <input
            type="text"
            name="name"
            value={name}
            onChange={e => handleNameUpdate(e)}
          />
        </label>
        <button onClick={e => handleFetchCharacters(e)}>Search</button>
      </form>
    </div>
  );
};

CharacterSearch.props = {
  name: PropTypes.string.isRequired,
  handleNameUpdate: PropTypes.func.isRequired,
  handleFetchCharacters: PropTypes.func.isRequired
};

export default CharacterSearch;
