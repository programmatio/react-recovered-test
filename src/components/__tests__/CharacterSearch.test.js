import React from 'react';
import { shallow } from 'enzyme';
import CharacterSearch from '../CharacterSearch';

const mockNameUpdate = jest.fn();
const mockHandleFetchCharacters = jest.fn();

const props = {
  name: '',
  handleNameUpdate: mockNameUpdate,
  handleFetchCharacters: mockHandleFetchCharacters,
};

const wrapper = shallow(<CharacterSearch {...props} />);

describe("<CharacterSearch />", () => {
  it("should match the snapshot", function() {
    expect(wrapper).toMatchSnapshot();
  });

  it("should trigger mock function onChange ", () => {
    wrapper.find('[type="text"]').simulate("change");
    expect(mockNameUpdate.mock.instances.length).toEqual(1);
  });

  it("should trigger mock function on submission", () => {
    wrapper.find("form").simulate("submit");
    expect(mockHandleFetchCharacters.mock.instances.length).toEqual(1);
  });

  it("should trigger mock function on button click", () => {
    wrapper.find("button").simulate("click");
    expect(mockHandleFetchCharacters.mock.instances.length).toEqual(2);
  });
});
