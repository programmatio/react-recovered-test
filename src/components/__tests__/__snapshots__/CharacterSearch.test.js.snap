// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CharacterSearch /> should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CharacterSearch
    handleFetchCharacters={[Function]}
    handleNameUpdate={[Function]}
    name=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <form
        onSubmit={[Function]}
>
        <label>
                Search Starwars Character by Name:
                <input
                        name="name"
                        onChange={[Function]}
                        type="text"
                        value=""
                />
        </label>
        <button
                onClick={[Function]}
        >
                Search
        </button>
</form>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label>
            Search Starwars Character by Name:
            <input
                        name="name"
                        onChange={[Function]}
                        type="text"
                        value=""
            />
</label>,
          <button
            onClick={[Function]}
>
            Search
</button>,
        ],
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Search Starwars Character by Name:",
              <input
                name="name"
                onChange={[Function]}
                type="text"
                value=""
/>,
            ],
          },
          "ref": null,
          "rendered": Array [
            "Search Starwars Character by Name:",
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "name": "name",
                "onChange": [Function],
                "type": "text",
                "value": "",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "label",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Search",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Search",
          "type": "button",
        },
      ],
      "type": "form",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <form
          onSubmit={[Function]}
>
          <label>
                    Search Starwars Character by Name:
                    <input
                              name="name"
                              onChange={[Function]}
                              type="text"
                              value=""
                    />
          </label>
          <button
                    onClick={[Function]}
          >
                    Search
          </button>
</form>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <label>
              Search Starwars Character by Name:
              <input
                            name="name"
                            onChange={[Function]}
                            type="text"
                            value=""
              />
</label>,
            <button
              onClick={[Function]}
>
              Search
</button>,
          ],
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Search Starwars Character by Name:",
                <input
                  name="name"
                  onChange={[Function]}
                  type="text"
                  value=""
/>,
              ],
            },
            "ref": null,
            "rendered": Array [
              "Search Starwars Character by Name:",
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "name": "name",
                  "onChange": [Function],
                  "type": "text",
                  "value": "",
                },
                "ref": null,
                "rendered": null,
                "type": "input",
              },
            ],
            "type": "label",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Search",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Search",
            "type": "button",
          },
        ],
        "type": "form",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
